tests/integration_tests/test_preprocessing.py

```diff
@@ -910,17 +913,84 @@ def test_handle_features_with_few_shot_prompt_config(backend, retrieval_kwargs,
 
     with context:
         backend = initialize_backend(backend)
-        handle_features_with_prompt_config(
+        dataset_cols = handle_features_with_prompt_config(
             config,
-            dataset_cols,
+            df,
             feature_configs,
             backend=backend,
             split_col=split_col,
         )
 
+        assert len(dataset_cols) == 1
+        assert input_feature_name in dataset_cols
+
         # Inspect the generated prompts
-        for prompt in dataset_cols[input_feature_name]:
+        col = backend.df_engine.compute(dataset_cols[input_feature_name])
+        for prompt in col:
             # input_feature_name and output_feature_name should be in the prompt because
             # labeled samples are provided by the context
             assert input_feature_name in prompt
             assert output_feature_name in prompt
+
+
+@pytest.mark.llm
+@pytest.mark.parametrize("backend", ["local", "ray"])
+def test_handle_features_with_prompt_config_multi_col(backend, ray_cluster_2cpu):
+    df = pd.DataFrame(
+        [
+            {
+                "instruction": "Name this province",
+                "country": "Canada",
+                "year": "1871",
+                "answer": "British Columbia",
+            },
+            {
+                "instruction": "Name this city",
+                "country": "France",
+                "year": "1789",
+                "answer": "Paris",
+            },
+            {
+                "instruction": "Name this country",
+                "country": "UK",
+                "year": "1057",
+                "answer": "Wales",
+            },
+        ]
+    )
+
+    config = {
+        "model_type": MODEL_LLM,
+        "model_name": "gpt2",
+        "input_features": [text_feature(name="question", encoder={"type": "passthrough"})],
+        "output_features": [text_feature(name="answer")],
+        "prompt": {
+            "template": "You are a helpful chatbot. USER: {instruction}: {country}, {year} ASSISTANT:",
```

Comment:
Could we also add a test for prompt templates within an input feature definition?

Reply:
This is supported, see my PR description above:

```
input_features:
    - name: context
      type: text
      preprocessing:
          prompt:
            template: The {color} {animal} jumped over the {size} {object}
```

Reply:
Yes, going to add that today.

Reply:
Nice, thanks!